battles/helpers.py

```diff
@@ -0,0 +1,67 @@
+from django.conf import settings
+
+from templated_email import send_templated_mail
+
+from battles.models import TrainerTeam
+
+
+# one-a-one figth
+def compare_hitpoints(pokemon_1, pokemon_2):
+    if pokemon_1.hitpoints > pokemon_2.hitpoints:
+        return pokemon_1
+    return pokemon_2
+
+
+def compare_attack_defense(pokemon_1, pokemon_2):
+    if pokemon_1.attack > pokemon_2.defense:
+        return pokemon_1
+    return pokemon_2
+
+
+def get_one_a_one_fight_winner(pokemon_1, pokemon_2):
+    if compare_attack_defense(pokemon_1, pokemon_2) == compare_attack_defense(pokemon_2, pokemon_1):
+        return compare_attack_defense(pokemon_1, pokemon_2)
+    return compare_hitpoints(pokemon_1, pokemon_2)
+
+
+# battle fight
+def get_battle_winner(pokemons_creator, pokemons_opponent):
+    one_a_one_results = []
+    for (pokemon_c, pokemon_o) in zip(pokemons_creator, pokemons_opponent):
+        if get_one_a_one_fight_winner(pokemon_c, pokemon_o) == pokemon_c:
+            one_a_one_results.append('creator')
+        else:
+            one_a_one_results.append('opponent')
+
+    if one_a_one_results.count('creator') > one_a_one_results.count('opponent'):
+        return 'creator'
+    return 'opponent'
+
+
+# send email
+def send_battle_result_email(battle):
+    trainers = [battle.trainer_creator, battle.trainer_opponent]
+    for trainer in trainers:
+        trainer_opponent = trainers[1] if trainer == trainers[0] else trainers[0]
+        trainer_team = [
+            TrainerTeam.objects.get(battle_related=battle, trainer=trainer).pokemon_1,
+            TrainerTeam.objects.get(battle_related=battle, trainer=trainer).pokemon_2,
+            TrainerTeam.objects.get(battle_related=battle, trainer=trainer).pokemon_3
+        ]
```

Comment:
@pamella In here, you are making 3 queries to the database, to get 3 attributes related to the same instance.
You could simplify all that making only one query like that:
```
trainer_team = TrainerTeam.objects.get(battle_related=battle, trainer=trainer)
trainer_team_pokemons = [trainer_team.pokemon_1, trainer_team.pokemon_2, trainer_team.pokemon_3]
```
Or even get all the fields at once like this:
```
trainer_team = TrainerTeam.objects.filter(battle_related=battle, trainer=trainer).values('pokemon_1', 'pokemon_2', 'pokemon_3')
```

Reply:
I didn't realize I could have done that way. Done! Thanks!

Reply:
No problem :)
You made it even simpler, congrats :tada: